### PR TITLE
Add GitOps tutorial

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Add a tutorial showing how you can use Kyma CLI and Flux to automatically deploy your Git repository resources on a cluster.

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/9401

### CODE REVIEW

**Review by pPrecel — CHANGES_REQUESTED** [review]
Really good work! Only one small thing to improve

**Review by pPrecel — DISMISSED** [review]

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+- brew
+- Git repository
+
+## Steps
+
+These sections will lead you through the whole installation, configuration, and synchronization process. You will first install k3d and create a cluster for your custom resources (CRs). Then, you will need to apply the necessary Custom Resource Definitions (CRDs) from Kyma to be able to create Functions and triggers. Finally, you will install Flux and authorize it with the `write` access to your GitHub repository in which you store the resource files. Flux will automatically synchronize any new changes pushed to your repository with your k3d cluster.
+
+### Install and configure a k3d cluster
+
+1. Install k3d using brew:
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:24]:
```suggestion
1. Install k3d using Homebrew:
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+This tutorial shows how you can automate the deployment of local Kyma resources on a cluster using the GitOps logic. You will use [Kyma CLI](https://kyma-project.io/docs/cli/) to create an inline Python Function with a trigger. You will later push both resources to a GitHub repository of your choice and set up a GitOps operator to monitor the given repository folder and synchronize any changes in it with your cluster. For the purpose of this tutorial, you will install and use the [Flux](https://docs.fluxcd.io/en/1.17.1/tutorials/get-started.html) GitOps operator and a lightweight [k3d](https://k3d.io/) cluster.
+
+> **TIP:** Although this tutorial uses Flux to synchronize Git resources with the cluster, you can use an alternative GitOps operator for this purpose, such as [Argo](https://argoproj.github.io/argo-cd/).
+
+## Prerequisites
+
+All you need before you start is to have the following:
+
+- CLI
+- brew
```
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply docs/serverless/08-07-sync-function-with-gitops.md:15]: Should we add a link here? 🤔

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+### Install and configure a k3d cluster
+
+1. Install k3d using brew:
+
+  ```bash
+  brew install k3d
+  ```
+
+2. Cluster a default k3d cluster with a single server node:
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:30]:
```suggestion
2. Create a default k3d cluster with a single server node:
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+  k3d cluster create {CLUSTER-NAME}
+  ```
+
+This command also sets your context to the newly created cluster. Run this command to display the cluster information:
+
+  ```bash
+  kubectl cluster-info
+  ```
+
+3. Apply the Kyma's `functions.serverless.kyma-project.io` and `triggers.eventing.knative.dev` CRDs from the sources in the [`kyma`](https://github.com/kyma-project/kyma/tree/master/resources/cluster-essentials/files) repository. You will need them to create the Function and Trigger CRs on the cluster.
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:42]:
```suggestion
3. Apply the `functions.serverless.kyma-project.io` and `triggers.eventing.knative.dev` CRDs from sources in the [`kyma`](https://github.com/kyma-project/kyma/tree/master/resources/cluster-essentials/files) repository. You will need them to create the Function and Trigger CRs on the cluster.
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+  ```bash
+  kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/master/resources/cluster-essentials/files/functions.serverless.crd.yaml && kubectl apply -f https://raw.githubusercontent.com/kyma-project/kyma/master/resources/cluster-essentials/files/triggers.eventing.knative.dev.crd.yaml
+  ```
+4. Run this command to make sure the CRs were applied:
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:47]:
```suggestion
4. Run this command to make sure the CRs are applied:
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+1. Install k3d using brew:
+
+  ```bash
+  brew install k3d
+  ```
+
+2. Cluster a default k3d cluster with a single server node:
+
+  ```bash
+  k3d cluster create {CLUSTER-NAME}
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:33]:
```suggestion
  k3d cluster create {CLUSTER_NAME}
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+  source:
+      sourceType: inline
+      sourcePath: {FULL_PATH_TO_WORKSPACE_FOLDER}
+  ```
+
+### Install and configure Flux
+
+You can now install the Flux operator, connect it with a specific Git repository folder, and authorize Flux to automatically pull changes from this repository folder and apply them on your cluster.
+
+1. Install flux:
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:90]:
```suggestion
1. Install Flux:
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+
+  ```bash
+  brew install fluxctl
+  ```
+
+2. Create a `flux` Namespace for the Flux operator's CRDs.
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:96]:
```suggestion
2. Create a `flux` Namespace for the Flux operator's CRDs:
```

Comment (klaudiagrz) [reply docs/serverless/08-07-sync-function-with-gitops.md:96]: You ended previous sentences with a colon so, for consistency sake, I'd use a colon in all the other steps as well.

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+3. Export details of your GitHub repository - its name, the account name and related e-mail address, and the folder to which you will push Function and Trigger CRs built from local sources. Flux will synchronize the cluster with the content of this folder on the `main` branch.
+
+  ```bash
+  export GH_USER="{USERNAME}"
+  export GH_REPO="{REPOSITORY_NAME}"
+  export GH_EMAIL="{EMAIL_OF_YOUR_GH_ACCOUNT}"
+  export GH_FOLDER="{GH_FOLDER_FOR_FUNCTION_RESOURCES}"
+  ```
+
+4. Run this command to apply CRDs of the Flux operator to the `flux` Namespace on your cluster.
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:111]:
```suggestion
4. Run this command to apply CRDs of the Flux operator to the `flux` Namespace on your cluster:
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+
+6. Run this command to copy the SSH key to the clipboard:
+
+  ```bash
+  fluxctl identity --k8s-fwd-ns flux | pbcopy
+  ```
+
+7. Go to **Settings** in your GitHub account:
+
+  ![GitHub account settings](./assets/settings.png)
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:148]: I really like the borders in your screenshots! ❤️ 👍

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+
+8. Go to the **SSH and GPG keys** section and select the **New SSH key** button.
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:150]:
```suggestion
8. Go to the **SSH and GPG keys** section and select the **New SSH key** button:
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 12]
```diff
@@ -0,0 +1,252 @@
...
+
+  ![Create a new SSH key](./assets/create-ssh-key.png)
+
+9. Provide the new key name, paste the previously copied SSH key, and confirm changes by selecting the **Add SSH Key** button.
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:154]:
```suggestion
9. Provide the new key name, paste the previously copied SSH key, and confirm changes by selecting the **Add SSH Key** button:
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 13]
```diff
@@ -0,0 +1,252 @@
...
+  git clone git@github.com:${GH_USER}/${GH_REPO}.git
+  ```
+
+2. Go to the repository folder:
+
+  ```bash
+  cd {LOCAL_GITHUB_REPOSITORY_FOLDER}
+  ```
+
+3. Run the `apply` Kyma CLI command to create a Function CR in the YAML format on your remote GitHub repository. This command will generate the output in the `my-function.yaml` file.
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:175]:
```suggestion
3. Run the `apply` Kyma CLI command to create a Function CR in the YAML format in your remote GitHub repository. This command will generate the output in the `my-function.yaml` file.
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 14]
```diff
@@ -0,0 +1,252 @@
...
+
+  ```bash
+  kyma apply function --filename {FULL_PATH_TO_LOCAL_WORKSPACE_FOLDER}/config.yaml --output yaml --dry-run > ./{GH_REPO}/${GH_FOLDER}/my-function.yaml
+  ```
+
+4. Push the local changes to the remote repository:
+
+  ```bash
+  git add .  # Stage changes for the commit
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:184]: Consider using a tab on these comments to move them to the right so as to make a straight column.

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 15]
```diff
@@ -0,0 +1,252 @@
...
+
+  ```bash
+  kubectl get triggers
+  ```
+
+You can see that Flux synchronized the resources and the new Trigger CR for the Function was added to your cluster.
+
+## Reverting feature
+
+Once you set it up, Flux will keep monitoring the given Git repository folder for any changes. If you modify the existing resources directly on the cluster, Flux will automatically revert these changes and update the given resource back its version on the `main` branch of the Git repository.
```
Comment (klaudiagrz) [inline docs/serverless/08-07-sync-function-with-gitops.md:252]:
```suggestion
Once you set it up, Flux will keep monitoring the given Git repository folder for any changes. If you modify the existing resources directly on the cluster, Flux will automatically revert these changes and update the given resource back to its version on the `main` branch of the Git repository.  
```

File: docs/serverless/08-07-sync-function-with-gitops.md (hunk 5 above)
Comment (klaudiagrz) [reply docs/serverless/08-07-sync-function-with-gitops.md:47]: I'm not sure about this one tbh. Shouldn't we use the present tense whenever it's possible? 🤔 
(applies to the whole document)

Comment (kazydek) [reply docs/serverless/08-07-sync-function-with-gitops.md:47]: It is difficult to avoid future tense everywhere, though.

**Review by klaudiagrz — DISMISSED** [review]
LGTM 🦊

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 16]
```diff
@@ -0,0 +1,252 @@
...
+1. Back in the terminal, clone this GitHub repository to your current workspace location:
+
+  ```bash
+  git clone git@github.com:${GH_USER}/${GH_REPO}.git
+  ```
+
+2. Go to the repository folder:
+
+  ```bash
+  cd {LOCAL_GITHUB_REPOSITORY_FOLDER}
```
Comment (pPrecel) [inline docs/serverless/08-07-sync-function-with-gitops.md:172]:
```suggestion
  cd ${GH_REPO}
```

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 17]
```diff
@@ -0,0 +1,252 @@
...
+  ```
+
+3. Run the `apply` Kyma CLI command to create a Function CR in the YAML format in your remote GitHub repository. This command will generate the output in the `my-function.yaml` file.
+
+  ```bash
+  kyma apply function --filename {FULL_PATH_TO_LOCAL_WORKSPACE_FOLDER}/config.yaml --output yaml --dry-run > ./${GH_FOLDER}/my-function.yaml
```
Comment (pPrecel) [inline docs/serverless/08-07-sync-function-with-gitops.md:178]:
```suggestion
  mkdir ${GH_FOLDER}
  kyma apply function --filename {FULL_PATH_TO_LOCAL_WORKSPACE_FOLDER}/config.yaml --output yaml --dry-run > ./${GH_FOLDER}/my-function.yaml
```

**Review by pPrecel — COMMENTED** [review]
;)

**Review by klaudiagrz — APPROVED** [review]
LGTM 🦊

File: docs/serverless/08-07-sync-function-with-gitops.md [hunk 18]
```diff
@@ -0,0 +1,272 @@
...
+## Steps
+
+These sections will lead you through the whole installation, configuration, and synchronization process. You will first install k3d and create a cluster for your custom resources (CRs). Then, you will need to apply the necessary Custom Resource Definitions (CRDs) from Kyma to be able to create Functions and triggers. Finally, you will install Flux and authorize it with the `write` access to your GitHub repository in which you store the resource files. Flux will automatically synchronize any new changes pushed to your repository with your k3d cluster.
+
+### Install and configure a k3d cluster
+
+1. Install k3d using Homebrew:
+
+  ```bash
+  brew install k3d
```
Comment (dbadura) [inline docs/serverless/08-07-sync-function-with-gitops.md:27]: Brew is mainly used in Macos. We should also provide command for instaling k3d on Linux.

**Review by klaudiagrz — DISMISSED** [review]

File: docs/serverless/08-07-sync-function-with-gitops.md (hunk 18 above)
Comment (kazydek) [reply docs/serverless/08-07-sync-function-with-gitops.md:27]: That is a broader topic than just adding a link to such instruction as this tutorial was only tested on macOS. We could extend it in the future for other operating systems, but that would apply to all other tutorials as this is the approach we normally have in Kyma. At this point, I can only specify here that this concern macOS by saying `Install k3d using Homebrew on macOS:`.

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**